raiden/raiden_event_handler.py

```diff
@@ -254,10 +254,15 @@ def handle_contract_send_secretreveal(
             raiden: RaidenService,
             channel_reveal_secret_event: ContractSendSecretReveal,
     ):
-        # LEFTODO: Supply a proper block id
+        # LEFTODO: Here we should normally get ContractSendChannelClose.triggered_by_block_hash
+        # but it may not be properly populated yet
+        if channel_reveal_secret_event.triggered_by_block_hash == bytes(0):
```

Comment:
For this file, the value of `given_block_identifier` should be the latest value in the chain state. We don't want to check state against the time which the event was emitted, but we want to use a fixed block to have a consistent view.

Reply:
Ignore the above. We talked in person and lefteris explained to me his rationale.

This is using the confirmed block known to the node as `given_block_identifier`, that block will be used for `validate_preconditions` to have a consistent view of the blockchain, the check is **not** intended to handle any race conditions, only logic bugs when the emitted event doesn't make sense. Race conditions will be checked by the `estimate_gas` call and the reason will be checked by the `validate_after_failure`.